src/host/handle_ring_buffer.h

```diff
@@ -55,6 +55,12 @@ namespace asynchost
           std::cerr << msg << std::endl << std::flush;
           throw std::logic_error(msg);
         });
+
+      DISPATCHER_SET_MESSAGE_HANDLER(
+        bp, AdminMessage::stopped, [](const uint8_t* data, size_t size) {
+          uv_stop(uv_default_loop());
```

Comment:
I believe that doing this as late as here gives the host sufficient time to print out all the enclave logs on shutdown.

Reply:
It is possible that worker threads in the enclave are still executing and writing outbound messages - we may drop those. But this should still fix the safety issue, since this `uv_stop` should always be called from the "host thread", whereas the signal could be handled on any.